src/libraries/Microsoft.PowerFx.Core/Types/Enums/EnumStoreBuilder.cs

```diff
@@ -327,56 +341,47 @@ internal EnumStoreBuilder WithDefaultEnums()
             return this;
         }
 
-        internal EnumStore Build()
+        internal IEnumStore Build()
         {
-            return EnumStore.Build(new List<EnumSymbol>(EnumSymbols));
+            return this;
         }
         #endregion
 
-        private Dictionary<string, DType> RegenerateEnumTypes()
+        public IEnumerable<EnumSymbol> EnumSymbols
```

Comment:
This is implementing IEnumStore. 
That interface forces a linear scan on every enum lookup. Maybe we need a more efficient lookup interface/

Reply:
ConcurrentDictionary<string, string> manages it all for us

Reply:
I'd be interested in knowing the perf results you can get with it in your tests